### PR TITLE
Adding geographiclib-cxx 1.51

### PR DESCRIPTION
**Pull request by cffk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cffk** [comment]
This is my first contribution to conda-forge and I regard myself as somewhat of a novice with conda and docker.  geographiclib-cxx should be a pretty clean C++ package since it has no dependencies and just builds a library and a few ools.  However, I noticed several warning messages in the build.  Probably these are innocuous.  Is there any configuration changes I can make so that these warnings don't get triggered?  (I got some warnings because my CMakeLists.txt and the conda build were working at cross-purposes to set RPATH.  I've commented out all of this in my CMakeLists.txt.)  Here are the warnings (somewhat truncated):

* All platforms:
  * WARNING conda_build.config:arch(292): Setting build arch. This is only useful when pretending to be on another arch
  * WARNING conda_build.metadata:ns_cfg(118): No numpy version specified in conda_build_config.yaml.
  * WARNING conda_build.build:bundle_conda(1627): Importing conda-verify failed.  Please be sure to test your packages.

* Windows:
  * WARNING: failed to get_linkages, codefile_type('C:/Miniconda/conda-bld/geographiclib-cpp_1610377597900/_h_env\Library\bin\RhumbSolve.exe')=EXEfile
  * WARNING: failed to get_linkages, codefile_type('C:/Miniconda/conda-bld/geographiclib-cpp_1610377597900/_h_env\Library\bin\msvcp140_2d.dll')=DLLfile
  * WARNING (geographiclib-cpp): dso library package conda-forge::vs2015_runtime-14.28.29325-h5e1d092_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)

* Linux:
  * useradd: warning: the home directory already exists.
  * Warning: 'conda-forge' already in 'channels' list, moving to the top
  * WARNING :: get_rpaths_raw()=['XX:YY'] and patchelf=[['XX', 'YY']] disagree for ZZ/CartConvert :: 
  * Warning: rpath XX is outside prefix WW (removing it)

Thanks for your help.

File: recipes/geographiclib-cpp/bld.bat [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+cmake -G "NMake Makefiles" ^
+    -DGEOGRAPHICLIB_LIB_TYPE:STRING=SHARED ^
+    -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+    -DCMAKE_BUILD_TYPE:STRING=Release ^
+    ..
+if errorlevel 1 exit 1
+
+nmake
+if errorlevel 1 exit 1
```
Comment (ocefpaf) [inline recipes/geographiclib-cpp/bld.bat:14]: Is there a ctest or make test we can run here?

File: recipes/geographiclib-cpp/build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+mkdir -p build && cd build
+
+cmake \
+    -DGEOGRAPHICLIB_LIB_TYPE:STRING=SHARED \
+    -DCMAKE_INSTALL_PREFIX:PATH=${PREFIX} \
+    -DCMAKE_BUILD_TYPE:STRING=Release \
+    ..
+
+make -j$CPU_COUNT
+make install
```
Comment (ocefpaf) [inline recipes/geographiclib-cpp/build.sh:11]: Same as https://github.com/conda-forge/staged-recipes/pull/13650/files#r555386426

**Review by ocefpaf — APPROVED** [review]
LTGM. A minor question regarding the tests.

File: recipes/geographiclib-cpp/bld.bat (hunk 1 above)
Comment (cffk) [reply recipes/geographiclib-cpp/bld.bat:14]: Done

File: recipes/geographiclib-cpp/build.sh (hunk 2 above)
Comment (cffk) [reply recipes/geographiclib-cpp/build.sh:14]: Done

**Pull request merged**